java/arcs/core/data/expression/Expression.kt

```diff
@@ -420,9 +449,34 @@ sealed class Expression<out T> {
         val function: GlobalFunction,
         val arguments: List<Expression<*>>
     ) : Expression<T>() {
-        override fun <Result> accept(visitor: Visitor<Result>): Result = visitor.visit(this)
+        override fun <Result, Context> accept(visitor: Visitor<Result, Context>, ctx: Context) =
+            visitor.visit(this, ctx)
         override fun toString() = this.stringify()
     }
+
+    /**
+     * Represents an expression that sorts a sequence to produce a new sequence. Note that this
+     * operator can be expensive, and must traverse its qualifying sequence entirely.
+     *
+     * [selectors] is a non-empty list of expressions that return values to be used for ordering.
+     */
+    data class OrderByExpression<T>(
+        override val qualifier: Expression<Sequence<Scope>>,
+        val selectors: List<Pair<Expression<Any>, Boolean>>
```

Comment:
Consider replacing a Pair with a data class for added naming?

data class Selector(selectingExpr: Expression<Any>, descending: Boolean) ?

Reply:
Done